blocks/theano_expressions.py

```diff
@@ -0,0 +1,18 @@
+"""Frequently used Theano expressions."""
+from theano import tensor
+
+
+def L2_norm(tensors):
+    """Computes the total L2 norm of a set of tensors.
+
+    Parameters
+    ----------
+    tensors : iterable of :class:`~theano.Variable`
+        The tensors.
+
+    """
+    flattened = [tensor.as_tensor_variable(t).flatten() for t in tensors]
+    flattened = [(t if t.ndim > 0 else tensor.alloc(t, 1))
+                 for t in flattened]
+    joined = tensor.join(0, *flattened)
```

Comment:
It worked on my examples, but you're right. I thought element-wise operations performed a `join`, but they actually perform a `as_tensor_variable` on the list, which does a `join` as well, but only after prepending an empty dimension. By the way, `t.dimshuffle('x')` should add a dimension in place instead of allocating new memory for it.

You might want to update the dosctring by the way; right now it suggests that the inputs need to be tensors, but you allow for numeric inputs as well (and use that in the tests). If you only want to support tensor variables, then remove the `as_tensor_variable` part and just change the tests to look like `L2_norm([x, y]).eval({x: 1, y: [1, 2]})`.

Otherwise, enough nitpicking and LGTM!